app/services/api_bridge/query.rb

```diff
@@ -33,17 +33,47 @@ class Query
 
     # create a new query instance with some default settings
     def initialize(url, facets=[], options={})
-      if ApiBridge.is_url?(url)
+      if is_url?(url)
         @url = url
         @facet_fields = facets
-        options = _remove_rows(options)
-        @options = ApiBridge.override_options(@@default_options, options)
+        options = remove_rows(options)
+        @options = override_options(@@default_options, options)
         @options["facet"] = facets
       else
         raise "Provided URL must be valid! #{url}"
       end
     end
 
+    # set the start index based on page, start, and num requested parameters
+    def calc_start(options)
+      # if start is specified by user then don't override with page
+      page = set_page(options["page"])
+      # remove page from options
+      options.delete("page")
+      if !options.key?("start")
+        # use the page and rows to set a start, use default is no num specified
+        num = options.key?("num") ? options["num"].to_i : @options["num"].to_i
+        options["start"] = get_start(page, num)
+      end
+      options
+    end
+
+    # create a version of the request parameters to manipulate
+    # without modifying the original set
+    def clone_and_stringify_options(aOpts)
```

Comment:
Note: Resolved by re-written code but I'll leave the note here fyi

I think we ought to avoid using this camel-case letter prefix for variables we're going to clone like this: https://github.com/rubocop-hq/ruby-style-guide#snake-case-symbols-methods-vars

I think `opts_orig` is more clear but also relatively brief. Or just leaving it as `opts` is fine since you have a different name for the derivative object.

Reply:
Yes, this was definitely leftover syntax from my previous programming job.  `aSomething` indicated a parameter being passed in which was then assigned to something else or immediately manipulated into something else.  `_something` was a private method, so it was considered code smell if you ever saw `service.notifier._something()` or the like.  Another noteworthy one that doesn't come up very often here since ruby is structured differently than javascript, was `onSomething()` which indicated a callback / block to execute.

These days I try to only use camel case when I'm programming with JavaScript.  But yeah, that's why I did change this in the modernized-code pass, because our style guide prefers snake case.